tester.go

```diff
@@ -227,8 +246,15 @@ func validateResponseHeaderPatterns(response *http.Response, patterns map[string
 		values, ok := response.Header[http.CanonicalHeaderKey(header)]
 		if !ok {
 			if expectedToMatch {
+				if expectFailure {
+					continue
+				}
 				errors = append(errors, fmt.Errorf("response header \"%s\" not found, expected to match pattern \"%s\"", header, pattern))
+
 			} else {
+				if expectFailure {
+					continue
+				}
 				errors = append(errors, fmt.Errorf("response header \"%s\" not found, expected to be present", header))
 			}
 			continue
```

Comment:
So I'm having a bit of problem sorting out if `expectFailure` logic works as expected. To elaborate..

First, the `expectFailure` check in this section of code can be lifted out like this:

```go
		values, ok := response.Header[http.CanonicalHeaderKey(header)]
		if !ok {
			if expectFailure {
				continue
			}

			if expectedToMatch {
				errors = append(errors, fmt.Errorf("response header \"%s\" not found, expected to match pattern \"%s\"", header, pattern))
			} else {
				errors = append(errors, fmt.Errorf("response header \"%s\" not found, expected to be present", header))
			}
			continue
```

What this means is: if a header does **not** exist in the response AND the test should fail, the outer `for` loop continues. However the expectation itself hasn't _actually_ been asserted.

Consider tests like:
```yaml
# expectedToMatch=true
  - description: 'HTTP GET - should match header doesn't match, assert test failure'
    request:
      scheme: 'http'
      path: '/get'
    response:
      headers:
        patterns:
          content-typo: "application/text" # response content-type is "application/json"
    shouldFail: true
# ^ we haven't actually asserted that the failure is due to the pattern not matching; because the failure was induced by the header not existing (due to a typo: `typo` instead of `type`)

# expectedToMatch=false
  - description: 'HTTP GET - notMatching header matches, assert test failure'
    request:
      scheme: 'http'
      path: '/get'
    response:
      headers:
        notMatching:
          content-typo: "application/json" # response content-type is "application/json"
    shouldFail: true
# ^ we haven't actually asserted that the failure is due to the pattern matching; because the failure was induced by the header not existing (due to a typo: `typo` instead of `type`)
```

I think similar to the regex compile error above, the `expectFailure -> continue` doesn't belong in this section yet?

```suggestion
		values, ok := response.Header[http.CanonicalHeaderKey(header)]
		if !ok {
			if expectedToMatch {
				errors = append(errors, fmt.Errorf("response header \"%s\" not found, expected to match pattern \"%s\"", header, pattern))
			} else {
				errors = append(errors, fmt.Errorf("response header \"%s\" not found, expected to be present", header))
			}
			continue
```